### DISCUSSION TITLE
Locale management with domain + subpath routing

### DISCUSSION THREAD

**Discussion by antoinelin** [vercel/next.js#25723] [Ideas]
### Describe the feature you'd like to request

Heya Vercel team 👋🏻 

I would like to use the locale subpath routing + domain routing to do so:
- `example.com` which has two locales : `en` and `en-ca`
- `example.fr` which has one locale : `fr-ca`

The issue I face ATM is that these routes are working:
- `example.com/fr-ca`
- `example.fr/en`, `example.fr/en-ca`
- `example.fr/fr-ca`

And SEO crawlers use them for indexation which could lead to duplicate content strikes etc.

### Describe the solution you'd like

Could it be possible to :

- Disable subpath for defaultLocale on related domain
- Disable locales for specific domain (or use `domains > locales` to explicitly tells Next.js to only use the provided locales for the related domain)

Example :
```
  i18n: {
    locales: ['en', 'en-ca', 'fr-ca'],
    defaultLocale: 'en',
    domains: [
      {
        domain: 'example.com',
        defaultLocale: 'en',
        locales: ['en-ca'],
        disabledLocales: ['fr-ca']
        disableDefaultLocaleSubpath: true
      },
      {
        domain: 'example.fr',
        defaultLocale: 'fr-ca',
        disabledLocales: ['en-ca', 'en']
        disableDefaultLocaleSubpath: true
      },
    ],
  }
  ```
  
  So that we would have:
  - `example.com/en` -> `404` or force redirection to `example.com` with `en` locale
  - `example.com/fr-ca` -> `404`
  - `example.fr/en` -> `404`

### Describe alternatives you've considered

- Deploy the same app multiple times with different ENVs so I can control which locales are allowed or not -> Not an ideal idea,
- Add disallow rules on `robots.txt` with subpath patterns like (for `example.fr`) :
```
User-agent: *
Allow: /
Disallow: /_next/
Disallow: /en-ca/
Disallow: /en/
```
- Use canonicals and hreflang the right way to tells Google to index the page we want -> But as far as I understand, canonicals are suggestions you want to show to Google. It is not an obligation unlike a block on a robot.txt for example. What makes that google decides or not to follow your suggestion according to what he notices
- Use 301 redirections -> I don't think it is a good solution to use too many 301 for this use case which could lead to internal mesh issues with crawlers.

**Comment by antoinelin** [comment]
I would be happy to submit a PR if needed

**Reply by danyasnov** [reply]
any updates? I'm stuck with similar issue

**Reply by antoinelin** [reply]
Nop :/ I had to create a second Next.js app to do so...
